bin/save_gradle_dependency_cache

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -eu
+
+# The key is shared with bin/restore_gradle_dependency_cache
+GRADLE_DEPENDENCY_CACHE_KEY="GRADLE_DEPENDENCY_CACHE"
+
+echo "Saving Gradle dependency cache..."
+
+mkdir -p "$GRADLE_RO_DEP_CACHE"
+
+# https://docs.gradle.org/current/userguide/dependency_resolution.html#sub:cache_copy
+# Gradle suggests removing the "*.lock" files and the `gc.properties` file before saving cache
+cp -r ~/.gradle/caches/modules-2 "$GRADLE_RO_DEP_CACHE" \
+    && find "$GRADLE_RO_DEP_CACHE" -name "*.lock" -type f -delete \
+    && find "$GRADLE_RO_DEP_CACHE" -name "gc.properties" -type f -delete
+
+# `save_cache` & `restore_cache` scripts only work if they are called from the same directory
+pushd "$GRADLE_RO_DEP_CACHE_BASE_FOLDER"
+# For now we are using a single key - we might expand on this later by using dependency catalog version
+save_cache "$GRADLE_RO_DEP_CACHE_FOLDER_NAME" "$GRADLE_DEPENDENCY_CACHE_KEY" --force
```

Comment:
Looking at [the CI logs](https://buildkite.com/automattic/android-dependency-catalog/builds/65#530b7a28-5910-4bd7-8998-2265db987a44/139-172) I was surprised by the fact the key is so "generic", whit which I mean it doesn't have a hash or anything else to distinguish it from previous builds. This explains it 👍

Reply:
> since the cache is generated from a dependency catalog, instead of a specific project.

That's actually pretty cool 🧐